src/controllers/authentication.js

```diff
@@ -200,13 +201,21 @@ authenticationController.login = function (req, res, next) {
 	} else if (loginWith.indexOf('username') !== -1 && !validator.isEmail(req.body.username)) {
 		continueLogin(req, res, next);
 	} else {
-		res.status(500).send('[[error:wrong-login-type-' + loginWith + ']]');
+		var err = '[[error:wrong-login-type-' + loginWith + ']]';
+
+		if (req.body.noscript === 'true') {
+			return helpers.noScriptErrors(req, res, err, 500);
+		}
+		res.status(500).send(err);
```

Comment:
Ideally you would pull it all out into `helpers.noScriptErrors`. Not sure if that's possible though.